### ISSUE TITLE
Unable to put a file to a directory in LocalFileSystem

### ISSUE THREAD

**Issue by xuepeixin** [fsspec/filesystem_spec#967]
I tried to copy a file to another directory with the put function of LocalFileSystem, the code is 
```
from fsspec.implementations.local import LocalFileSystem

fs = LocalFileSystem()
fs.mkdir('/tmp/test_dir0')
fs.mkdir('/tmp/test_dir1')
with open('/tmp/test_dir0/a', 'w') as fp:
    fp.write('test')
fs.put('/tmp/test_dir0/a', '/tmp/test_dir1/')
```
but I get an Error traceback like this:
```Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/anaconda3/envs/py38/lib/python3.8/site-packages/fsspec/spec.py", line 823, in put
    self.put_file(lpath, rpath, **kwargs)
  File "/home/user/anaconda3/envs/py38/lib/python3.8/site-packages/fsspec/implementations/local.py", line 127, in put_file
    return self.cp_file(path1, path2, **kwargs)
  File "/home/user/anaconda3/envs/py38/lib/python3.8/site-packages/fsspec/implementations/local.py", line 117, in cp_file
    shutil.copyfile(path1, path2)
  File "/home/user/anaconda3/envs/py38/lib/python3.8/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
IsADirectoryError: [Errno 21] Is a directory: '/tmp/test_dir1'
```

However the doc of put tell me that 
> If rpath ends with a "/", it will be assumed to be a directory, and target files will go within.

I guess directory is supported by design, so is it a bug?

**Comment by martindurant** [comment]
I wonder, does `cp` work? For LocalFileSytem, that should be the same thing (which is why it isn't tested).

**Comment by martindurant** [comment]
Would you consider making a PR in which 
```
put = cp
```
forLocalFileSystem to simplify things?

**Comment by xuepeixin** [comment]
Ok, I will try.

**Comment by ianthomas23** [comment]
This was fixed by #1148. Test code from above:
```python
from fsspec.implementations.local import LocalFileSystem

fs = LocalFileSystem()
fs.mkdir('/tmp/test_dir0')
fs.mkdir('/tmp/test_dir1')
with open('/tmp/test_dir0/a', 'w') as fp:
    fp.write('test')
fs.put('/tmp/test_dir0/a', '/tmp/test_dir1/')
print(fs.find('/tmp/test_dir1'))
```
produces output
```
['/tmp/test_dir1/a']
```
as expected.

**Comment by martindurant** [comment]
Nice to see all these closed!
